src/manifest/owned/pfm.rs

```diff
@@ -102,6 +81,49 @@ pub enum Element {
     },
 }
 
+/// asdf
+#[allow(missing_docs)]
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct AllowableFw {
+    #[cfg_attr(
+        feature = "serde",
+        serde(deserialize_with = "crate::serde::de_radix")
+    )]
+    pub version_count: u8,
+    #[cfg_attr(
+        feature = "serde",
+        serde(
+            deserialize_with = "crate::serde::de_bytestring",
+            serialize_with = "crate::serde::se_bytestring",
+        )
+    )]
+    pub firmware_id: Vec<u8>,
+    #[cfg_attr(
+        feature = "serde",
+        serde(
+            deserialize_with = "crate::serde::de_radix",
+            serialize_with = "crate::serde::se_bin",
+        )
+    )]
+    flags: u8,
```

Comment:
Unfortunately, this means that this can't be constructed by external crates now... not sure what the best answer here is unfortunately. We could add a `new` function that takes all the parameters... but then I'd insist on accessors for the two other fields (which we'd then make private.)

Reply:
Why can't other crates construct them? 

We could have a function on the `Element`, but that's always going to yield an awkward experience. This example sets the flags, if its an `Element::AllowableFw` variant:

```
impl Element {
    fn set_update_takes_effect(&self, update_takes_effect: UpdateTakesEffect) -> bool {
    if let Element::AllowableFw  {
       version_count,
       firmware_id,
       mut ref flags,
     }  = self {
        *flags = update_takes_effect as u8;
         true
   } else {
         false
  }
}
```

But that's not great because its weird if you don't have an `AllowableFw`. Thoughts?